<a href="https://colab.research.google.com/github/KASHYAPCHETAN438/DSA/blob/main/AlgorithmsAndImplementation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Sample of Algorithms and Implementation

01. Even or Odd
02. Max of two numbers
03. Min of two numbers
04. Max of three numbers
05. Min of three numbers
06. Max of four numbers
07. Min of four numbers
08. Max of five numbers
09. Min of five numbers
10. Swaping of two integer values
11. Absolute value
12. Sum of n natural numbers
13. Factorial of given number
14. Digits Extraction
15. Count digits
16. Reverse a number
17. Paliandrome number
18. Trailing Zeros of factorial
19. x to the power y
20. Sum of digits
21. Sum of even digits
22. Sum of odd digits
23. Sum of prime digits
24. Prime Number or Not
25. All divisors of N
26. Perfect Number
27. armstrong number
28. strong number
29. Fib sequence
30. Trib Sequence


--------

## Problem Statement -01

### Write a program to check whether a given number is Even or Odd.  

### Example  

For number 7:  
- 7 % 2 = 1 → Odd  

Final output → Odd  

### Algorithm  


1. Start  
2. Read a number `n` from the user  
3. Check if `n % 2 == 0`  
   - If yes → Print "Even"  
   - Else → Print "Odd"  
4. End  
  


In [ ]:
num=int(input("Enter the number"))
if num%2==0:
  print(num,"is even")
else:
  print(num,"is odd")

Enter the number7
7 is odd



------

## Problem Statement -02

### Write a program to find the maximum of two numbers.  

### Example  

For numbers 10 and 20:  
- Max(10, 20) = 20  

Final output → 20  

### Algorithm  

1. Start  
2. Read two numbers `a` and `b` from the user  
3. Compare `a` and `b`  
   - If `a > b` → Print `a` as maximum  
   - Else → Print `b` as maximum  
4. End  


In [ ]:
# Problem 02: Max of two numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))

if a > b:
    print(f"Maximum = {a}")
else:
    print(f"Maximum = {b}")


Enter first number: 10
Enter second number: 2
Maximum = 10


-------
## Problem Statement -03

### Write a program to find the minimum of two numbers.  

### Example  

For numbers 10 and 20:  
- Min(10, 20) = 10  

Final output → 10  

### Algorithm  

1. Start  
2. Read two numbers `a` and `b` from the user  
3. Compare `a` and `b`  
   - If `a < b` → Print `a` as minimum  
   - Else → Print `b` as minimum  
4. End  


In [ ]:
# Problem 03: Min of two numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))

if a < b:
    print(f"Minimum = {a}")
else:
    print(f"Minimum = {b}")


Enter first number: 4
Enter second number: 8
Minimum = 4


------

## Problem Statement -04

### Write a program to find the maximum of three numbers.  

### Example  

For numbers 10, 25, 15:  
- Max(10, 25, 15) = 25  

Final output → 25  

### Algorithm  

1. Start  
2. Read three numbers `a`, `b`, `c` from the user  
3. Compare `a`, `b`, and `c`  
   - If `a > b` and `a > c` → Print `a` as maximum  
   - Else if `b > a` and `b > c` → Print `b` as maximum  
   - Else → Print `c` as maximum  
4. End  


In [ ]:
# Problem 04: Max of three numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))
c = int(input("Enter third number: "))

max_val = a
if b > max_val:
    max_val = b
if c > max_val:
    max_val = c

print(f"Maximum = {max_val}")


Enter first number: 4
Enter second number: 8
Enter third number: 1
Maximum = 8


----

## Problem Statement -05

### Write a program to find the minimum of three numbers.  

### Example  

For numbers 10, 25, 15:  
- Min(10, 25, 15) = 10  

Final output → 10  

### Algorithm  

1. Start  
2. Read three numbers `a`, `b`, `c` from the user  
3. Compare `a`, `b`, and `c`  
   - If `a < b` and `a < c` → Print `a` as minimum  
   - Else if `b < a` and `b < c` → Print `b` as minimum  
   - Else → Print `c` as minimum  
4. End  


In [ ]:
# Problem 05: Min of three numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))
c = int(input("Enter third number: "))

min_val = a
if b < min_val:
    min_val = b
if c < min_val:
    min_val = c

print(f"Minimum = {min_val}")


Enter first number: 6
Enter second number: 8
Enter third number: 120
Minimum = 6


-----

## Problem Statement -06

### Write a program to find the maximum of four numbers.  

### Example  

For numbers 12, 7, 19, 3:  
- Max(12, 7, 19, 3) = 19  

Final output → 19  

### Algorithm  

1. Start  
2. Read four numbers `a`, `b`, `c`, `d` from the user  
3. Initialize `max = a`  
4. Compare `max` with `b`, `c`, `d` one by one  
   - If any number is greater than `max`, update `max`  
5. Print `max` as maximum  
6. End  


In [ ]:
# Problem 06: Max of four numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))
c = int(input("Enter third number: "))
d = int(input("Enter fourth number: "))

max_val = a
for num in [b, c, d]:
    if num > max_val:
        max_val = num

print(f"Maximum = {max_val}")


Enter first number: 78
Enter second number: 95
Enter third number: 15
Enter fourth number: 20
Maximum = 95


----

## Problem Statement -07

### Write a program to find the minimum of four numbers.  

### Example  

For numbers 12, 7, 19, 3:  
- Min(12, 7, 19, 3) = 3  

Final output → 3  

### Algorithm  

1. Start  
2. Read four numbers `a`, `b`, `c`, `d` from the user  
3. Initialize `min = a`  
4. Compare `min` with `b`, `c`, `d` one by one  
   - If any number is smaller than `min`, update `min`  
5. Print `min` as minimum  
6. End  


In [ ]:
# Problem 07: Min of four numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))
c = int(input("Enter third number: "))
d = int(input("Enter fourth number: "))

min_val = a
for num in [b, c, d]:
    if num < min_val:
        min_val = num

print(f"Minimum = {min_val}")


Enter first number: 78
Enter second number: 52
Enter third number: 26
Enter fourth number: 14
Minimum = 14


-----
## Problem Statement -08

### Write a program to find the maximum of five numbers.  

### Example  

For numbers 4, 18, 9, 23, 11:  
- Max(4, 18, 9, 23, 11) = 23  

Final output → 23  

### Algorithm  

1. Start  
2. Read five numbers `a`, `b`, `c`, `d`, `e` from the user  
3. Initialize `max = a`  
4. Compare `max` with `b`, `c`, `d`, `e` one by one  
   - If any number is greater than `max`, update `max`  
5. Print `max` as maximum  
6. End  


In [ ]:
# Problem 08: Max of five numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))
c = int(input("Enter third number: "))
d = int(input("Enter fourth number: "))
e = int(input("Enter fifth number: "))

max_val = a
for num in [b, c, d, e]:
    if num > max_val:
        max_val = num

print(f"Maximum = {max_val}")


Enter first number: 24
Enter second number: 65
Enter third number: 78
Enter fourth number: 25
Enter fifth number: 45
Maximum = 78


------
## Problem Statement -09

### Write a program to find the minimum of five numbers.  

### Example  

For numbers 4, 18, 9, 23, 11:  
- Min(4, 18, 9, 23, 11) = 4  

Final output → 4  

### Algorithm  

1. Start  
2. Read five numbers `a`, `b`, `c`, `d`, `e` from the user  
3. Initialize `min = a`  
4. Compare `min` with `b`, `c`, `d`, `e` one by one  
   - If any number is smaller than `min`, update `min`  
5. Print `min` as minimum  
6. End  


In [ ]:
# Problem 09: Min of five numbers
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))
c = int(input("Enter third number: "))
d = int(input("Enter fourth number: "))
e = int(input("Enter fifth number: "))

min_val = a
for num in [b, c, d, e]:
    if num < min_val:
        min_val = num

print(f"Minimum = {min_val}")


Enter first number: 32
Enter second number: 69
Enter third number: 75
Enter fourth number: 25
Enter fifth number: 259
Minimum = 25


--------
## Problem Statement -10

### Write a program to swap two integer values.  

### Example  

For values a = 5, b = 7:  
- After swap: a = 7, b = 5  

Final output → a = 7, b = 5  

### Algorithm  

1. Start  
2. Read two integers `a` and `b` from the user  
3. Use a temporary variable `temp`  
   - `temp = a`  
   - `a = b`  
   - `b = temp`  
4. Print `a` and `b` after swapping  
5. End  


In [ ]:
a,b=5,7
temp=a
a=b
b=temp
print(f"Value of a is : {a} \nValue of b is :{b}")

Value of a is : 7 
Value of b is :5


In [ ]:
# without third varible

a,b=5,7
a=a+b     # a--> 5+7 = 12  ,b=7
b=a-b     # b--> 12-7 = 5
a=a-b     # a--> 12-5 = 7
print(f"Value of a is : {a} \nValue of b is :{b}")




Value of a is : 7 
Value of b is :5


------

## Problem Statement -11

### Write a program to find the absolute value of a number.  

### Example  

For number -25:  
- Absolute value = 25  

Final output → 25  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. If `n < 0` → `n = -n`  
4. Print `n` as absolute value  
5. End  


In [ ]:
n=int(input("Enter the Negative number :"))

if n<0:
  print("Abouslte value is : ",n*-1)
else:
   print("Abouslte value is : ",n)

Enter the Negative number :-5
Abouslte value is :  5


In [ ]:
n=int(input("Enter the Negative number :"))
print("Abouslte value is : ",abs(n))

Enter the Negative number :-6
Abouslte value is :  6


--------
## Problem Statement -12

### Write a program to find the sum of n natural numbers.  

### Example  

For n = 5:  
- Sum = 1 + 2 + 3 + 4 + 5 = 15  

Final output → 15  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `sum = 0`  
4. For i = 1 to n → Add i to sum  
5. Print sum  
6. End  


In [ ]:
n=int(input("Enter the number where you want to sum :"))
sum=0
for i in range(1,n+1):
 sum+=i
print(f"Sum of first {n} natural number is :{sum}")

Enter the number where you want to sum :5
Sum of first 5 natural number is :15


--------
## Problem Statement -13

### Write a program to find the factorial of a given number.  

### Example  

For number 5:  
- 5! = 5 × 4 × 3 × 2 × 1 = 120  

Final output → 120  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `fact = 1`  
4. For i = 1 to n → `fact = fact * i`  
5. Print `fact`  
6. End  


In [ ]:
n=int(input("Enter the number :"))
fact=1
for i in range(1,n+1):
  fact*=i
print("Factroial of ",n,"is :",fact)

Enter the number :5
Factroial of  5 is : 120


In [ ]:
n=int(input("Enter the number :"))
fact=1
for i in range(1,n+1):         # n=0 loop is not execute then it print fact initil value (fact=1)
  fact*=i
print("Factroial of ",n,"is :",fact)

Enter the number :0
Factroial of  0 is : 1


-----
## Problem Statement -14

### Write a program to extract the digits present in a given number.  

### Example  

For number 4725:  
- Digits = 4, 7, 2, 5  

Final output → 4 7 2 5  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. While `n > 0` →  
   - digit = n % 10  
   - Store digit in a list or print it  
   - n = n / 10  
4. Print digits in correct order  
5. End  


In [ ]:
# Problem 14: Digits Extraction
n = int(input("Enter a number: "))
digits = []
temp = n

while temp > 0:
    digit = temp % 10
    digits.insert(0, digit)  # maintain order
    temp //= 10

print("Digits:", digits)


Enter a number: 4725
Digits: [4, 7, 2, 5]


-----
## Problem Statement -15

### Write a program to count the number of digits in a given number.  

### Example  

For number 4725:  
- Count = 4  

Final output → 4  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `count = 0`  
4. While `n > 0` →  
   - n = n / 10  
   - Increment count by 1  
5. Print count  
6. End  


In [ ]:

n = int(input("Enter a number: "))
count = 0
temp = abs(n)

if temp == 0:
    count = 1
else:
    while temp > 0:
        temp //= 10
        count += 1

print(f"Number of digits = {count}")


Enter a number: 4725
Number of digits = 4


--------
## Problem Statement -16

### Write a program to find the reverse of a given number.  

### Example  

For number 4725:  
- Reverse = 5274  

Final output → 5274  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `rev = 0`  
4. While `n > 0` →  
   - digit = n % 10  
   - rev = rev * 10 + digit  
   - n = n / 10  
5. Print rev  
6. End  


In [ ]:
# Problem 16: Reverse a number
n = int(input("Enter a number: "))
rev = 0
temp = n

while temp > 0:
    digit = temp % 10
    rev = rev * 10 + digit
    temp //= 10

print(f"Reverse = {rev}")


Enter a number: 4725
Reverse = 5274


-----
## Problem Statement -17

### Write a program to check whether the given number is a palindrome or not.  

### Example  

For number 121:  
- Reverse = 121 → same as original → Palindrome  

Final output → Palindrome  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Store original number in `temp`  
4. Initialize `rev = 0`  
5. While `n > 0` →  
   - digit = n % 10  
   - rev = rev * 10 + digit  
   - n = n / 10  
6. If `rev == temp` → Print "Palindrome"  
   Else → Print "Not Palindrome"  
7. End  


In [ ]:
# Problem 17: Palindrome number
n = int(input("Enter a number: "))
rev = 0
temp = n

while temp > 0:
    digit = temp % 10
    rev = rev * 10 + digit
    temp //= 10

if rev == n:
    print(f"{n} is Palindrome")
else:
    print(f"{n} is Not Palindrome")


Enter a number: 121
121 is Palindrome


--------
## Problem Statement -18

### Write a program to find the number of trailing zeros in the factorial of a number.  

### Example  

For number 10:  
- 10! = 3628800 → Trailing zeros = 2  

Final output → 2  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `count = 0`  
4. While n >= 5 →  
   - count = count + n/5  
   - n = n / 5  
5. Print count  
6. End  


In [ ]:
import math
npt=int(input("Enter the number"))
count=0
math.factorial(npt)

while n>=5:
  count=count+n/5
  n=n/5
else:
  print(count)


Enter the number5


NameError: name 'n' is not defined

In [ ]:
import math

n = int(input("Enter a number: "))
fact = math.factorial(n)
print(fact)

Enter a number: 5
120


In [ ]:

fact_str = str(fact)

print(fact_str)

a=reversed(fact_str)
print(a)

120


In [ ]:


# Count trailing zeros
count = 0
for ch in reversed(fact_str):
    if ch == '0':
        count += 1
    else:
        break

print(f"Trailing zeros = {count}")


In [ ]:
n = int(input("Enter a number: "))
count = 0
i = 5

while n // i >= 1:
    count += n // i
    i *= 5

print(f"Trailing zeros = {count}")


Enter a number: 10
Trailing zeros = 2


In [ ]:
a=set()
print(a)

set()


-----
## Problem Statement -19

### Write a program to compute x to the power y.  

### Example  

For x = 2, y = 5:  
- 2⁵ = 32  

Final output → 32  

### Algorithm  

1. Start  
2. Read numbers `x` and `y` from the user  
3. Initialize `result = 1`  
4. For i = 1 to y →  
   - result = result * x  
5. Print result  
6. End  


In [1]:

x = int(input("Enter the base (x): "))
y = int(input("Enter the exponent (y): "))


result = 1


for i in range(1, y + 1):
    result = result * x


print(f"{x}^{y} = {result}")


Enter the base (x): 5
Enter the exponent (y): 2
5^2 = 25


------
## Problem Statement -20

### Write a program to find the sum of digits of a given number.  

### Example  

For number 4725:  
- Sum = 4 + 7 + 2 + 5 = 18  

Final output → 18  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `sum = 0`  
4. While `n > 0` →  
   - digit = n % 10  
   - sum = sum + digit  
   - n = n / 10  
5. Print sum  
6. End  


In [2]:
# Program to find the sum of digits of a number


n = int(input("Enter a number: "))

# Initialize sum
sum_digits = 0

# Process each digit
while n > 0:
    digit = n % 10          # Get the last digit
    sum_digits += digit     # Add it to sum
    n = n // 10             # Remove the last digit


print("Sum of digits =", sum_digits)


Enter a number: 54
Sum of digits = 9


In [ ]:
# Problem 20: Sum of digits
n = int(input("Enter a number: "))
sum_digits = 0
temp = abs(n)

while temp > 0:
    digit = temp % 10
    sum_digits += digit
    temp //= 10

print(f"Sum of digits = {sum_digits}")


Enter a number: 4725
Sum of digits = 18


-------
## Problem Statement -21

### Write a program to find the sum of even digits of a given number.  

### Example  

For number 4725:  
- Even digits = 4, 2  
- Sum = 6  

Final output → 6  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `sum = 0`  
4. While `n > 0` →  
   - digit = n % 10  
   - If digit % 2 == 0 → sum = sum + digit  
   - n = n / 10  
5. Print sum  
6. End  


----------
## Problem Statement -22

### Write a program to find the sum of odd digits of a given number.  

### Example  

For number 4725:  
- Odd digits = 7, 5  
- Sum = 12  

Final output → 12  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `sum = 0`  
4. While `n > 0` →  
   - digit = n % 10  
   - If digit % 2 != 0 → sum = sum + digit  
   - n = n / 10  
5. Print sum  
6. End  


In [4]:
# Program to find the sum of even digits of a number


n = int(input("Enter a number: "))


sum_even = 0

# Process each digit
while n > 0:
    digit = n % 10           # Get the last digit
    if digit % 2 == 0:       # Check if it is even
        sum_even += digit
    n = n // 10              # Remove the last digit


print("Sum of even digits =", sum_even)


Enter a number: 48238
Sum of even digits = 22


------
## Problem Statement -23

### Write a program to find the sum of prime digits of a given number.  

### Example  

For number 4725:  
- Prime digits = 2, 5, 7  
- Sum = 14  

Final output → 14  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `sum = 0`  
4. While `n > 0` →  
   - digit = n % 10  
   - If digit is prime (2, 3, 5, 7) → sum = sum + digit  
   - n = n / 10  
5. Print sum  
6. End  


In [5]:

n = int(input("Enter a number: "))


sum_prime = 0

prime_digits = {2, 3, 5, 7}

# Process each digit
while n > 0:
    digit = n % 10              # Get the last digit
    if digit in prime_digits:   # Check if digit is prime
        sum_prime += digit
    n = n // 10                 # Remove the last digit


print("Sum of prime digits =", sum_prime)


Enter a number: 7541741
Sum of prime digits = 19


---------
## Problem Statement -24

### Write a program to check whether the given number is prime or not.  

### Example  

For number 7:  
- Divisors = 1, 7 → Prime  

Final output → Prime  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. If n <= 1 → Print "Not Prime"  
4. Else →  
   - For i = 2 to sqrt(n)  
      - If n % i == 0 → Print "Not Prime" and exit  
5. If loop completes → Print "Prime"  
6. End  


In [6]:

import math


n = int(input("Enter a number: "))

# Check for numbers less than or equal to 1
if n <= 1:
    print("Not Prime")
else:
    # Assume number is prime
    is_prime = True

    # Check divisibility from 2 to sqrt(n)
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            is_prime = False
            break

    # Print result
    if is_prime:
        print("Prime")
    else:
        print("Not Prime")


Enter a number: 774
Not Prime


-------
## Problem Statement -25

### Write a program to find all divisors of a given number N.  

### Example  

For N = 12:  
- Divisors = 1, 2, 3, 4, 6, 12  

Final output → 1 2 3 4 6 12  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. For i = 1 to n →  
   - If n % i == 0 → Print i  
4. End  


In [8]:

n = int(input("Enter a number: "))

print("Divisors of", n, "are:")

# Loop from 1 to n
for i in range(1, n + 1):
    if n % i == 0:
        print(i, end=" ")


Enter a number: 12
Divisors of 12 are:
1 2 3 4 6 12 

-----
## Problem Statement -26

### Write a program to check whether the given number is a Perfect Number.  

### Example  

For number 28:  
- Divisors = 1, 2, 4, 7, 14  
- Sum = 28 → Perfect Number  

Final output → Perfect Number  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Initialize `sum = 0`  
4. For i = 1 to n-1 →  
   - If n % i == 0 → sum = sum + i  
5. If sum == n → Print "Perfect Number"  
   Else → Print "Not Perfect Number"  
6. End  


In [9]:

n = int(input("Enter a number: "))

# Initialize sum of divisors
sum_divisors = 0

# Loop to find divisors from 1 to n-1
for i in range(1, n):
    if n % i == 0:
        sum_divisors += i

# Check if sum of divisors equals the number
if sum_divisors == n:
    print("Perfect Number")
else:
    print("Not Perfect Number")


Enter a number: 28
Perfect Number


-------
## Problem Statement -27

### Write a program to check whether the given number is an Armstrong Number.  

### Example  

For number 153:  
- 1³ + 5³ + 3³ = 153 → Armstrong  

Final output → Armstrong Number  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Store original number in `temp`  
4. Initialize `sum = 0`  
5. While n > 0 →  
   - digit = n % 10  
   - sum = sum + digit³  
   - n = n / 10  
6. If sum == temp → Print "Armstrong Number"  
   Else → Print "Not Armstrong Number"  
7. End  


In [10]:

n = int(input("Enter a number: "))

temp = n


sum_digits = 0

# Process each digit
while n > 0:
    digit = n % 10               # Get the last digit
    sum_digits += digit ** 3     # Add cube of digit
    n = n // 10                  # Remove the last digit


# Check if sum equals original number
if sum_digits == temp:
    print("Armstrong Number")
else:
    print("Not Armstrong Number")


Enter a number: 153
Armstrong Number


----------
## Problem Statement -28

### Write a program to check whether the given number is a Strong Number.  

### Example  

For number 145:  
- 1! + 4! + 5! = 145 → Strong Number  

Final output → Strong Number  

### Algorithm  

1. Start  
2. Read a number `n` from the user  
3. Store original number in `temp`  
4. Initialize `sum = 0`  
5. While n > 0 →  
   - digit = n % 10  
   - factorial = 1  
   - For i = 1 to digit → factorial = factorial * i  
   - sum = sum + factorial  
   - n = n / 10  
6. If sum == temp → Print "Strong Number"  
   Else → Print "Not Strong Number"  
7. End  


In [11]:

n = int(input("Enter a number: "))


temp = n

# Initialize sum
sum_factorials = 0

# Process each digit
while n > 0:
    digit = n % 10
    factorial = 1

    # Calculate factorial of the digit
    for i in range(1, digit + 1):
        factorial *= i

    sum_factorials += factorial
    n = n // 10

# Check if sum of factorials equals original number
if sum_factorials == temp:
    print("Strong Number")
else:
    print("Not Strong Number")


Enter a number: 145
Strong Number


--------
## Problem Statement -29

### Write a program to generate Fibonacci sequence up to n terms.  

### Example  

For n = 6:  
- Sequence = 0, 1, 1, 2, 3, 5  

Final output → 0 1 1 2 3 5  

### Algorithm  

1. Start  
2. Read number `n` from the user  
3. Initialize `a = 0`, `b = 1`  
4. Print `a` and `b`  
5. For i = 3 to n →  
   - c = a + b  
   - Print c  
   - a = b, b = c  
6. End  


In [12]:

n = int(input("Enter the number of terms: "))


a, b = 0, 1

# Check for n <= 0
if n <= 0:
    print("Please enter a positive integer")
elif n == 1:
    print("Fibonacci sequence:", a)
else:
    print("Fibonacci sequence:", a, b, end=" ")

    # Generate remaining terms
    for i in range(3, n + 1):
        c = a + b
        print(c, end=" ")
        a, b = b, c


Enter the number of terms: 6
Fibonacci sequence: 0 1 1 2 3 5 

--------
## Problem Statement -30

### Write a program to generate Tribonacci sequence up to n terms.  

### Example  

For n = 6:  
- Sequence = 0, 1, 1, 2, 4, 7  

Final output → 0 1 1 2 4 7  

### Algorithm  

1. Start  
2. Read number `n` from the user  
3. Initialize `a = 0`, `b = 1`, `c = 1`  
4. Print `a`, `b`, `c`  
5. For i = 4 to n →  
   - d = a + b + c  
   - Print d  
   - a = b, b = c, c = d  
6. End  


In [13]:

n = int(input("Enter the number of terms: "))


a, b, c = 0, 1, 1

# Handle cases for n <= 0
if n <= 0:
    print("Please enter a positive integer")
elif n == 1:
    print("Tribonacci sequence:", a)
elif n == 2:
    print("Tribonacci sequence:", a, b)
elif n == 3:
    print("Tribonacci sequence:", a, b, c)
else:
    print("Tribonacci sequence:", a, b, c, end=" ")

    # Generate remaining terms
    for i in range(4, n + 1):
        d = a + b + c
        print(d, end=" ")
        a, b, c = b, c, d


Enter the number of terms: 6
Tribonacci sequence: 0 1 1 2 4 7 